R/interact.R

```diff
@@ -124,8 +124,27 @@ step_interact_new <-
 ## one large set of collected terms.
 #' @export
 prep.step_interact <- function(x, training, info = NULL, ...) {
+
+  # Empty selection
+  if (identical(x$terms[[1]], quo())) {
+    return(
+        step_interact_new(
+        terms = x$terms,
+        role = x$role,
+        trained = TRUE,
+        objects = x$objects,
+        sep = x$sep,
+        skip = x$skip,
+        id = x$id
+      )
+    )
+  }
+
   # Identify any selectors that are involved in the interaction
   # formula
+  tmp_terms <- as.formula(rlang::as_label(x$terms[[1]]))
+  environment(tmp_terms) <- environment(x$terms[[1]])
+  x$terms <- tmp_terms
```

Comment:
These lines of code switches the list of quosures back into the formula the rest of the function expects